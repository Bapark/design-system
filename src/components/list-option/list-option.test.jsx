import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import ListOption from './list-option.jsx';

describe('src/components/list-option/list-option', () => {
  it('renders defaults', () => {
    const tree = renderer.create(<ListOption />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('children API', () => {
    it('accepts a node as children', () => {
      render(<ListOption><div><span>Text is here!</span></div></ListOption>);
      expect(screen.getByText('Text is here!')).toBeInTheDocument();
    });
  });

  describe('selected API', () => {
    it('is selected when selected is true', () => {
      render(<ListOption selected>SelectMe!</ListOption>);
      expect(screen.getByText('SelectMe!')).toHaveAttribute('aria-selected', 'true');
    });

    it('is not selected when selected is false', () => {
      render(<ListOption>SelectMe!</ListOption>);
      expect(screen.getByText('SelectMe!')).toHaveAttribute('aria-selected', 'false');
    });
  });

  describe('title API', () => {
    it('accepts a title', () => {
      const title = "Hello. Is it me you're looking for?";
      render(<ListOption title={title} />);

      expect(screen.getByRole('option')).toHaveAttribute('title', title);
    });
  });
});
