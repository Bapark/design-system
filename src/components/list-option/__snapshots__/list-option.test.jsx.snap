// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/list-option/list-option renders defaults 1`] = `
<li
  aria-selected={false}
  className="option"
  role="option"
  tabIndex="0"
/>
`;
